ethcore/evm/src/interpreter/mod.rs

```diff
@@ -589,9 +595,15 @@ impl<Cost: CostType> Interpreter<Cost> {
 				Self::copy_data_to_memory(&mut self.mem, stack, params.code.as_ref().map_or_else(|| &[] as &[u8], |c| &**c as &[u8]));
 			},
 			instructions::EXTCODECOPY => {
+				use std::ops::Deref;
+
 				let address = u256_to_address(&stack.pop_back());
 				let code = ext.extcode(&address)?;
-				Self::copy_data_to_memory(&mut self.mem, stack, &code);
+				Self::copy_data_to_memory(
+					&mut self.mem,
+					stack,
+					code.as_ref().map(|c| &c.deref()[..]).unwrap_or(&[])
```

Comment:
I think `code.as_ref().map(|c| &(*c)[..]).unwrap_or(&[])` should compile. Not sure if that's preferable though.